subprojects/core/src/main/java/org/gradle/api/internal/file/copy/FilterChain.java

```diff
@@ -109,20 +111,24 @@ public void add(final Closure closure) {
         add(new ClosureBackedTransformer(closure));
     }
 
-    public void expand(final Map<String, ?> properties) {
+    public void expand(final Map<String, ?> properties, final Provider<Boolean> escapeBackslash) {
         transformers.add(new Transformer<Reader, Reader>() {
             @Override
             public Reader transform(Reader original) {
                 try {
                     Template template;
                     try {
                         SimpleTemplateEngine engine = new SimpleTemplateEngine();
+                        if (escapeBackslash.isPresent()) {
```

Comment:
I think we just want to call get on the Provider here, otherwise you get an inconsistent behavior where if you clear `escapeBackslash`, you get the default behavior of SimpleTemplateEngine and not the default behavior of Gradle (which happen to be the same right now).

This effects `DefaultFileCopyDetails.expand(Map)` above, but I would just have it delegate to `expand(Map,Action)` like you're doing in other places with a `Actions.doNothing()`.

Reply:
I wasn't sure about this but as we are changing things there anyway I agree that we can force `false` by default regardless of SimpleTemplateEngine changes.